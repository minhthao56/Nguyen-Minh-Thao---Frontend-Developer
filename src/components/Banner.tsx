export default function Banner() {
  return (
    <div className="bg-banner bg-cover bg-no-repeat h-96 items-center flex flex-col justify-center">
      <h1 className="text-white font-bold text-4xl">SHOP SPACEX</h1>
      <p className="text-white font-thin">
        Lorem ipsum dolor sit amet consectetur, adipisicing elit.
      </p>
    </div>
  );
}
