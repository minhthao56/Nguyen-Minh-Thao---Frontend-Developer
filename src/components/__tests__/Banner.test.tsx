import { render, screen } from "@testing-library/react";
import Banner from "../Banner";

describe("<Banner/>", () => {
  const renderComponent = () => {
    return render(<Banner />);
  };

  it("should match to snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should has necessary text", () => {
    renderComponent();

    expect(
      screen.getByText(
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit."
      )
    ).toBeInTheDocument();

    expect(screen.getByText("SHOP SPACEX")).toBeInTheDocument();
  });
});
