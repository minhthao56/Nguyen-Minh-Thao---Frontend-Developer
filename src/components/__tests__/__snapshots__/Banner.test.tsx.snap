// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner/> should match to snapshot 1`] = `
<div>
  <div
    class="bg-banner bg-cover bg-no-repeat h-96 items-center flex flex-col justify-center"
  >
    <h1
      class="text-white font-bold text-4xl"
    >
      SHOP SPACEX
    </h1>
    <p
      class="text-white font-thin"
    >
      Lorem ipsum dolor sit amet consectetur, adipisicing elit.
    </p>
  </div>
</div>
`;
